File: demo-java/src/main/java/com/example/demojava/components/Badge.java
package com.example.demojava.components;

import blynx.thymeleaf.compositiondialect.CompositionComponent;
import blynx.thymeleaf.compositiondialect.CompositionComponentContext;

public class Badge extends CompositionComponent {

    public String type;

    public Badge(CompositionComponentContext context) {
        super(context);
        Object typeAttr = context.getAttributes().get("type");
        this.type = typeAttr != null ? typeAttr.toString() : null;
    }

    public String classNames() {
        return "badge " + ("danger".equals(this.type) ? "bg-danger" : "bg-secondary");
    }
}

File: demo/src/main/kotlin/com/example/demo/components/Badge.kt
package com.example.demo.components

import blynx.thymeleaf.compositiondialect.CompositionComponent
import blynx.thymeleaf.compositiondialect.CompositionComponentContext

class Badge(context: CompositionComponentContext) : CompositionComponent(context) {
    var type: String?

    init {
        this.type = context.attributes["type"].toString()
    }

    public fun classNames(): String {
        return "badge " + (if (this.type == "danger") "bg-danger" else "bg-secondary")
    }
}
